src/tests/multihost/alltests/test_cache_performance.py

```diff
@@ -0,0 +1,1305 @@
+                     'userPassword': [b'Secret123'],
+                     'loginShell': [b'/bin/bash']}
+
+        dn = f'uid=test{count}_foo{i},ou=People,dc=example,dc=test'
+        ldif_writer.unparse(dn, user_info)
+
+    for i in range(1, count):
+        memberdn = bytes(f'uid=test{count}_foo1,ou=People,dc=example,dc=test', encoding='utf-8')
+        group_info = {'cn': [bytes(f'test{count}_ldapusers{i}', encoding='utf-8')],
+                      'gidNumber': [bytes(f'{UID+i}', encoding='utf-8')],
+                      'objectClass': [b'top', b'posixGroup', b'groupOfNames'],
+                      'member': [memberdn]}
+
+        group_dn = f'cn=test{count}_ldapusers{i},ou=Groups,dc=example,dc=test'
+        if i <= 3:
+            for j in range(2, count):
+                user_dn = bytes(f'uid=test{count}_foo{j},ou=People,dc=example,dc=test', encoding='utf-8')
+                group_info['member'].append(user_dn)
+        else:
+            for j in range(2, 3):
+                user_dn = bytes(f'uid=test{count}_foo{j},ou=People,dc=example,dc=test', encoding='utf-8')
+                group_info['member'].append(user_dn)
+
+        ldif_writer.unparse(group_dn, group_info)
+
+    multihost.master[0].transport.put_file("/tmp/group.ldif", "/tmp/group.ldif")
+    ldap_cmd = f"ldapadd -x -H ldap://{multihost.master[0].sys_hostname} -D 'cn=Directory Manager' -w 'Secret123' \
+                -f /tmp/group.ldif"
+    cmd1 = multihost.master[0].run_command(ldap_cmd, raiseonerr=False)
+    time.sleep(5)
+    trun_cmd = "truncate -s 0 /tmp/group.ldif"
+    cmd2 = multihost.master[0].run_command(trun_cmd, raiseonerr=False)
+    fh.truncate(0)
+    fh.close()
+    assert cmd1.returncode == 0, f'{ldap_cmd} did not execute successfully'
+    assert cmd2.returncode == 0, f'{trun_cmd} did not execute successfully'
+
+
+def disable_dslimit(multihost):
+    """It Disables limits in Directory Server to fetch user / group records."""
+
+    content = 'dn: cn=config\nchangetype: modify\nadd: nsslapd-sizelimit\nnsslapd-sizelimit: -1\n\
+\ndn: cn=config,cn=ldbm database,cn=plugins,cn=config\nchangetype: modify\nreplace: nsslapd-lookthroughlimit\n\
+nsslapd-lookthroughlimit: -1\n\
+\ndn: cn=config,cn=ldbm database,cn=plugins,cn=config\nchangetype: modify\nreplace: nsslapd-idlistscanlimit\n\
+nsslapd-idlistscanlimit: -1'
+    multihost.client[0].put_file_contents("/root/ds_disable_limit.ldif", content)
+    ldap_cmd = f'ldapmodify -x -H ldap://{multihost.master[0].sys_hostname} \
+                   -D "cn=Directory Manager" -w "Secret123" -f /root/ds_disable_limit.ldif'
+    cmd1 = multihost.client[0].run_command(ldap_cmd, raiseonerr=False)
+    dirsrv_cmd = "dsctl example1 restart"
+    cmd2 = multihost.master[0].run_command(dirsrv_cmd, raiseonerr=False)
+    assert cmd1.returncode == 0, f'{ldap_cmd} did not execute successfully'
+    assert cmd2.returncode == 0, f'{dirsrv_cmd} did not execute successfully'
+
+
+def user_auth(multihost, user):
```

Comment:
Why is there another function for ssh authentication? We got multiple of them already:
https://github.com/SSSD/sssd/blob/master/src/tests/multihost/sssd/testlib/common/utils.py#L594
and
https://github.com/SSSD/sssd/blob/master/src/tests/multihost/sssd/testlib/common/utils.py#L742

Reply:
The test cases ranging from 19-24, required to list the home directory from a specific user, the above functions help in performing ssh authentication but do not list the files.
In all other places, I can replace ssh authentication with above methods.

Reply:
Why did the list the home? Is there some rhyme and reason behind that? You are not collecting the homedir listing nor using it in any way.

Reply:
Ok, explained enough. Feel free to keep the custom function.